Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/SendMessageJob.hs

```diff
@@ -0,0 +1,86 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module SharedLogic.SendMessageJob where
+
+import qualified Data.Map as M
+import qualified Domain.Types.Message.Message as Domain
+import qualified Domain.Types.Message.MessageReport as Domain
+import qualified Domain.Types.Message.MessageTranslation as D
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto (EsqDBReplicaFlow)
+import qualified Kernel.Storage.Esqueleto as Esq
+import Kernel.Streaming.Kafka.Commons (KafkaTopic)
+import Kernel.Streaming.Kafka.Producer (produceMessage)
+import Kernel.Streaming.Kafka.Producer.Types (KafkaProducerTools)
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Lib.Scheduler
+import Lib.Scheduler.JobStorageType.DB.Queries (createJobIn)
+import SharedLogic.Allocator (SchedulerJobType (..), SendDriverMessagesJobData (..))
+import qualified Storage.Queries.Message.MessageReport as MRQuery
+import qualified Storage.Queries.Message.MessageTranslation as MTQuery
+
+sendMessagesToDrivers ::
+  ( EsqDBReplicaFlow m r,
+    EsqDBFlow m r,
+    Log m,
+    HasField "broadcastMessageTopic" r KafkaTopic,
+    HasField "kafkaProducerTools" r KafkaProducerTools
+  ) =>
+  Job 'SendDriverMessages ->
+  m ExecutionResult
+sendMessagesToDrivers Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  let jobData = jobInfo.jobData
+  let (batchIds, restIds) = splitAt 30 jobData.driversIds -- TODO: make batch size configurable
```

Comment:
and why you didn't  it?

Reply:
Because i was asked to finish it asap. if this wont be merged in the beginning of the next week i will finish this as well